tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_provision.c

```diff
@@ -172,6 +208,81 @@ static void test_provisioner_pb_adv_multi(void)
 	PASS();
 }
 
+/** @brief Verify that the provisioner can provision a device multiple times after resets
+ */
+static void test_provisioner_pb_adv_reprovision(void)
+{
+	int err;
+
+	k_sem_init(&prov_sem, 0, 1);
+
+	bt_mesh_device_setup();
+
+	err = bt_mesh_cdb_create(test_net_key);
+	ASSERT_OK(err, "Failed to create CDB (err %d)\n", err);
+
+	err = bt_mesh_provision(test_net_key, 0, 0, 0, 0x0001, dev_key);
+	ASSERT_OK(err, "Provisioning failed (err %d)", err);
+
+	for (int i = 0; i < PROV_REPROV_COUNT; i++) {
+		LOG_INF("Provisioner prov loop #%d, waiting for prov ...\n", i);
+		ASSERT_OK(k_sem_take(&prov_sem, K_SECONDS(20)),
+			"Provisioner reprovision #%d fail", i);
+
+		uint8_t status;
+		size_t subs_count = 1;
+		uint16_t sub;
+		struct bt_mesh_cfg_mod_pub healthpub = { 0 };
+
+		/* Check that publication and subscription are reset after last iteration */
+		err = bt_mesh_cfg_mod_sub_get(0, prov_addr - 1, prov_addr - 1,
+					      BT_MESH_MODEL_ID_HEALTH_SRV, &status, &sub,
+					      &subs_count);
+		ASSERT_OK(err, "Sub get failed (err %d)", err);
+		ASSERT_OK(status, "Sub get did not return success (status = 0x%x)", status);
+		ASSERT_TRUE(subs_count == 0, "Subscriptions not cleared");
+
+		err = bt_mesh_cfg_mod_pub_get(0, prov_addr - 1, prov_addr - 1,
+					      BT_MESH_MODEL_ID_HEALTH_SRV, &healthpub, &status);
+		ASSERT_OK(err, "Pub get failed (err %d)", err);
+		ASSERT_OK(status, "Pub get did not return success (status = 0x%x)", status);
+		ASSERT_TRUE(healthpub.addr == BT_MESH_ADDR_UNASSIGNED, "Pub not cleared");
+
+
+		/* Set pub and sub to check that they are reset */
+		healthpub.addr = 0xc001;
+		healthpub.app_idx = 0;
+		healthpub.cred_flag = false;
+		healthpub.ttl = 10;
+		healthpub.period = BT_MESH_PUB_PERIOD_10SEC(1);
+		healthpub.transmit = BT_MESH_TRANSMIT(3, 100);
+
+		err = bt_mesh_cfg_app_key_add(0, prov_addr - 1, 0, 0, test_app_key, &status);
+		ASSERT_OK(err, "AppKey add failed (err %d)", err);
```

Comment:
Please note that `ASSERT_OK` doesn't accept message:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bluetooth/bsim_bt/bsim_test_mesh/src/mesh_test.h#L47-L55

Reply:
I see this done everywhere in the rest of the file. Should I clean up all of them?

Reply:
Or make `ASSERT_OK` take message as an argument.